test/smoke-fails/dyn_libs/Makefile

```diff
@@ -0,0 +1,28 @@
+include ../../Makefile.defs
+
+TESTNAME        = dyn_libs
+TESTSRC_MAIN    = main.cpp
+TESTSRC_AUX     =
+TESTSRC_ALL     = $(TESTSRC_MAIN) $(TESTSRC_AUX)
+
+CLANG           ?= clang++
+OMP_BIN         = $(AOMP)/bin/$(CLANG)
+CC              = $(OMP_BIN) $(VERBOSE)
+EXTRA_LDFLAGS   = -L. -lA -lB
+EXTRA_OMP_FLAGS =
+
+libA.so: libA.cpp libA.h Makefile.dynlibs
+	AOMP_GPU=$(AOMP_GPU) make -f Makefile.dynlibs libA.so
+
+libB.so: libB.cpp libB.h Makefile.dynlibs
+	AOMP_GPU=$(AOMP_GPU) make -f Makefile.dynlibs libB.so
+
+clean ::
+	make -f Makefile.dynlibs clean
+
+include ../Makefile.rules
+
+# override default run rule
+run:
+	HSA_XNACK=1 LD_LIBRARY_PATH=${PWD} ./dyn_libs
```

Comment:
You would want to put this into a `RUNCMD` variable, which is picked up by the Makefile machinery automatically.
Otherwise the special rule will not be picked up the `make check`, which executes in `check_smoke.sh`.

Reply:
Also, the `HSA_XNACK` and `LD_LIBRARY_PATH` setting should go into `RUNENV` and the Makefile should require a GFX that is capable of USM/XNACK.
You can do this now by setting
```
SUPPORTED = $(SUPPORTS_USM)
```